pkg/virtctl/create/vm/vm.go

```diff
@@ -290,6 +301,48 @@ func (c *createVM) newVM() *v1.VirtualMachine {
 	}
 }
 
+func (c *createVM) addDiskWithBootOrder(flag string, vm *v1.VirtualMachine, name string, bootOrder *uint) error {
+	if bootOrder != nil {
+		if *bootOrder == 0 {
+			return params.FlagErr(flag, "bootorder must be greater than 0")
+		}
+
+		if _, ok := c.bootOrders[*bootOrder]; ok {
+			return params.FlagErr(flag, "bootorder %d was specified multiple times", *bootOrder)
+		}
+
+		vm.Spec.Template.Spec.Domain.Devices.Disks = append(vm.Spec.Template.Spec.Domain.Devices.Disks, v1.Disk{
+			Name:      name,
+			BootOrder: bootOrder,
+		})
+
+		c.bootOrders[*bootOrder] = name
+	}
+
+	return nil
+}
+
+// getInferFromVolume returns the volume to infer the instancetype or preference from.
+// It returns either the disk with the lowest boot order or the first volume in the VM spec.
+func (c *createVM) getInferFromVolume(flag string, vm *v1.VirtualMachine) (string, error) {
```

Comment:
IMO it is in scope because without #9631 we always want the infer flags to infer from the volume the VM booted from. Without this method this isn't possible.